addons/flannel/template/base/install.sh

```diff
@@ -80,3 +91,83 @@ function flannel_weave_conflict() {
 function flannel_antrea_conflict() {
     ls /etc/cni/net.d/*antrea* >/dev/null 2>&1
 }
+
+function flannel_is_single_node() {
+    local nodecount=
+    nodecount=$(kubectl get nodes --no-headers | wc -l)
+    if [ "$nodecount" -eq 1 ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+function weave_to_flannel() {
+    local dst="$DIR/kustomize/flannel"
+
+    echo "REMOVING WEAVE"
+    remove_weave
+
+    echo "UPDATING KUBEADM FOR FLANNEL"
+    flannel_kubeadm
+
+    echo "APPLYING FLANNEL"
+    kubectl -n kube-flannel apply -k "$dst/"
+
+    sleep 60
+    echo "RESTARTING KUBELET"
+    sudo systemctl stop kubelet
+    sudo iptables -t nat -F && sudo iptables -t mangle -F && sudo iptables -F && sudo iptables -X
+    sudo systemctl restart containerd
+    sudo systemctl start kubelet
+
+    sleep 60
+    echo "RESTARTING kube-system"
+    kubectl -n kube-system delete pods --all
+    kubectl -n kube-flannel delete pods --all
+
+    sleep 60
+    echo "RESTARTING CSI"
+    kubectl -n longhorn-system delete pods --all || true
+    kubectl -n rook-ceph delete pods --all || true
+    kubectl -n openebs delete pods --all || true
+
+    sleep 60
+    echo "RESTARTING ALL OTHER PODS"
+    local ns=
+    for ns in $(kubectl get ns -o name | grep -Ev '(kube-system|longhorn-system|rook-ceph|openebs|kube-flannel)' | cut -f2 -d'/'); do
+        kubectl delete pods -n "$ns" --all
+    done
+    sleep 60
+}
+
+function remove_weave() {
+    # firstnode only
+    kubectl -n kube-system delete daemonset weave-net
+    kubectl -n kube-system delete rolebinding weave-net
+    kubectl -n kube-system delete role weave-net
+    kubectl delete clusterrolebinding weave-net
+    kubectl delete clusterrole weave-net
+    kubectl -n kube-system delete serviceaccount weave-net
+    kubectl -n kube-system delete secret weave-passwd
+
+    # all nodes
+    sudo rm -f /opt/cni/bin/weave-*
+    sudo rm -rf /etc/cni/net.d
+    sudo ip link delete weave
+
+}
+
+function flannel_kubeadm() {
+    # search for 'serviceSubnet', add podSubnet above it
+    local pod_cidr_range_line=
+    pod_cidr_range_line="  podSubnet: ${POD_CIDR}"
+    if grep 'podSubnet:' /opt/replicated/kubeadm.conf; then
+        sed -i "s_  podSubnet:.*_${pod_cidr_range_line}_" /opt/replicated/kubeadm.conf
+    else
+        sed -i "/serviceSubnet/ s/.*/${pod_cidr_range_line}\n&/" /opt/replicated/kubeadm.conf
+    fi
+
+    sudo kubeadm init phase upload-config kubeadm --config=/opt/replicated/kubeadm.conf
+    sudo kubeadm init phase control-plane controller-manager --config=/opt/replicated/kubeadm.conf
```

Comment:
```suggestion
    kubeadm init phase upload-config kubeadm --config=/opt/replicated/kubeadm.conf
    kubeadm init phase control-plane controller-manager --config=/opt/replicated/kubeadm.conf
```

Reply:
🤦 how on earth did I leave that in there